File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/presentation/base/BaseContractMvpView.java
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.base;

public interface BaseContractMvpView<P extends BaseContractMvpPresenter> {

    void hideProgressDialog();
    void showProgressDialog(String msg);
    void showProgressDialog(String msg, Boolean cancelable, Boolean cancelableInTouchMode);
    void showError(String exception);
    void attachPresenter(P basePresenter);
}

File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/presentation/base/BaseContractMvpView.kt
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.base

interface BaseContractMvpView {

    fun hideProgressDialog()
    fun showProgressDialog(msg: String)
    fun showProgressDialog(msg: String, cancelable: Boolean, cancelableInTouchMode: Boolean)
    fun showError(exception: String)
}
